src/modules/dataman/dataman.c

```diff
@@ -949,7 +949,15 @@ _file_initialize(unsigned max_offset)
 		close(dm_operations_data.file.fd);
 
 		if (incompat) {
+#ifdef CONFIG_ARCH_BOARD_AEROCORE2
```

Comment:
@smithandrewc - As I understand it, this code ensures that the file format is compatible, with, the version that wrote.  Other platforms (px4fmuv2) use the FRAM as well for dataman storage, So I do not understand the comment or the need for the ifdef.  

Also as a rule we do not want want to have any ```CONFIG_ARCH_BOARD_``` in the system files. If this ifdef has a logical purpose (which is still TBD) we would want other boards to be able use the feature as BOARD_xxx. Like the naming that is done  https://github.com/PX4/Firmware/blob/master/src/drivers/boards/common/board_common.h or things like https://github.com/PX4/Firmware/blob/master/src/drivers/boards/tap-v1/board_config.h#L193 and https://github.com/PX4/Firmware/blob/master/src/drivers/boards/tap-v1/board_config.h#L245 

If you would help me to understand the usage I can suggest a name and where is should be placed.